pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,10 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// ExternalBuildStrategy defines input parameters specific to an External build.
+type ExternalBuildStrategy struct {
```

Comment:
In previous conversations we were not creating os builds & were only going to update build config when external builds ran (think this was @smarterclayton's proposal). It does seem to make more sense to create os builds when triggering a job. This would need to be hooked in with the external job to create on pre build, update status on post build, with hopefully hooks somehow along the pipeline to update build status as it goes. Considering there will be two sources to trigger builds (os api & Jenkins itself) where should this sync be?

Options i can think of are:

* Logic in Jenkins: Regardless of trigger source, Jenkins plugin pre/post build steps create build & keep it updated via workflow steps. Os builder simply starts Jenkins build, exits, or just tails Jenkins build logs until completion.
* Logic in builder pod: If build started by os then os create build, start Jenkins job, keep in sync in builder pod. If build started by Jenkins, Jenkins triggers creation of builder pod on pre build with URL of job for builder pod to watch & sync.

I think the first option is more complete (probably able to more easily hook in with pipeline steps to update build status).